src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs

```diff
@@ -1121,18 +1121,16 @@ private void ImportDataMembers()
                             DataMember memberContract = new DataMember(member);
 
                             memberContract.Name = DataContract.EncodeLocalName(member.Name);
-                            object[] optionalFields = null;
-                            if (optionalFields == null || optionalFields.Length == 0)
```

Comment:
> As we are going to add OptionalFieldAttribute in Core

Yes, it's been added here:
https://github.com/dotnet/corefx/blob/dev/api/src/System.Runtime/ref/System.Runtime.cs#L5469
along with the rest of the runtime serialization primitives.  Do you have an issue open on GitHub to add back such pieces of code that were removed?